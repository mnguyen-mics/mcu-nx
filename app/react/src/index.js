import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import configureStore from './store';
import { Navigator } from './containers/Navigator';

const store = configureStore();

const micsProvider = (
  <Provider store={store}>
    <Navigator />
  </Provider>
);

render(
  micsProvider,
  document.getElementById('mcs-react-app') // eslint-disable-line no-undef
);
