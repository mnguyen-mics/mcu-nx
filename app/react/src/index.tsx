/* eslint-disable */
import React from 'react';
import { render } from 'react-dom';

import App from './App';

declare global {
  interface Window {
    require: any;
    mainRev: any;
  }
}

render(<App />, document.getElementById('mcs-react-app'), () => {
  window.require([window.mainRev], () => {
    //
  });
});
