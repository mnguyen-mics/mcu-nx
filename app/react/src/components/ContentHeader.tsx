import * as React from 'react';

interface Props {
  title: React.ReactNode;
  subTitle?: React.ReactNode;
  loading?: boolean;
  size?: 'large' | 'medium';
}

class ContentHeader extends React.Component<Props> {
  render() {
    const { title, subTitle, loading, size } = this.props;

    const content = (
      <div>
        <div className={`content-subtitle`}>{subTitle}</div>
        <div
          className={size === 'medium' ? `content-title-2` : `content-title`}
        >
          {title}
        </div>
      </div>
    );

    return (
      <div className="content-header">
        {loading ? <i className="mcs-table-cell-loading-large" /> : content}
      </div>
    );
  }
}

export default ContentHeader;
