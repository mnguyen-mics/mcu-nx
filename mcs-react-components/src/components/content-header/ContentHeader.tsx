import * as React from 'react';

export interface ContentHeaderProps {
  title: React.ReactNode;
  subTitle?: React.ReactNode;
  loading?: boolean;
  size?: 'large' | 'medium';
}

class ContentHeader extends React.Component<ContentHeaderProps> {
  render() {
    const { title, subTitle, loading, size } = this.props;

    const content = (
      <div>
        <div className="mcs-contentHeader_subtitle">{subTitle}</div>
        <div
          className={`mcs-contentHeader_title--${size}`}
        >
          {title}
        </div>
      </div>
    );

    return (
      <div className="mcs-contentHeader">
        {loading ? <i className="mcs-table-cell-loading-large" /> : content}
      </div>
    );

    
  }
}

export default ContentHeader;
