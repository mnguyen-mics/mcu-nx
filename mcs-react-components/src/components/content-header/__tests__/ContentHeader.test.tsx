import * as React from 'react';
import { MemoryRouter } from 'react-router';
import { IntlProvider } from 'react-intl';
import * as TestRenderer from 'react-test-renderer';

import ContentHeader, { ContentHeaderProps } from '../ContentHeader';

it('renders a medium content header', () => {
  const props: ContentHeaderProps = {
    title: 'Titre',
    size: 'medium'
  };

  const component = TestRenderer.create(
    <IntlProvider locale="en">
      <MemoryRouter>
        <ContentHeader {...props} />
      </MemoryRouter>
    </IntlProvider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a large content header', () => {
  const props: ContentHeaderProps = {
    title: 'Titre',
    size: 'large'
  };

  const component = TestRenderer.create(
    <IntlProvider locale="en">
      <MemoryRouter>
        <ContentHeader {...props} />
      </MemoryRouter>
    </IntlProvider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the "evolved" content header', () => {
  const props: ContentHeaderProps = {
    title: 'Titre',
    subTitle: 'SubTitre',
    loading: false,
  };

  const component = TestRenderer.create(
    <IntlProvider locale="en">
      <MemoryRouter>
        <ContentHeader {...props} />
      </MemoryRouter>
    </IntlProvider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the loading content header', () => {
  const props: ContentHeaderProps = {
    title: 'Titre',
    subTitle: 'SubTitre',
    loading: true,
  };

  const component = TestRenderer.create(
    <IntlProvider locale="en">
      <MemoryRouter>
        <ContentHeader {...props} />
      </MemoryRouter>
    </IntlProvider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
