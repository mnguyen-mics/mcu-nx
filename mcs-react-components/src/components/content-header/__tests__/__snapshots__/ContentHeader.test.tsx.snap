// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a large content header 1`] = `
<div
  className="mcs-contentHeader"
>
  <div>
    <div
      className="mcs-contentHeader_subtitle"
    />
    <div
      className="mcs-contentHeader_title--large"
    >
      Titre
    </div>
  </div>
</div>
`;

exports[`renders a medium content header 1`] = `
<div
  className="mcs-contentHeader"
>
  <div>
    <div
      className="mcs-contentHeader_subtitle"
    />
    <div
      className="mcs-contentHeader_title--medium"
    >
      Titre
    </div>
  </div>
</div>
`;

exports[`renders the "evolved" content header 1`] = `
<div
  className="mcs-contentHeader"
>
  <div>
    <div
      className="mcs-contentHeader_subtitle"
    >
      SubTitre
    </div>
    <div
      className="mcs-contentHeader_title--undefined"
    >
      Titre
    </div>
  </div>
</div>
`;

exports[`renders the loading content header 1`] = `
<div
  className="mcs-contentHeader"
>
  <i
    className="mcs-table-cell-loading-large"
  />
</div>
`;
